src/utils/colonyClient.ts

```diff
@@ -0,0 +1,21 @@
+import { AnyColonyClient } from '@colony/colony-js';
+
+import networkClient from '~networkClient';
+
+const colonyClientsCache: Record<string, AnyColonyClient> = {};
+
+/**
+ * Method attempting to return a cached colony client,
+ * or get one from the networkClient if there's no cached entry
+ */
+export const getCachedColonyClient = async (
```

Comment:
This is a very good start and I like your approach.

But for future improvements I would extend this to also be able to fetch/cache all types of contract clients: network, colony, tokens, but more important extensions.

Reply:
Thanks for the suggestions. I thought about doing it in this PR, but here's what I'm proposing we do instead:
- Network client - for now, it looks like `networkClient.ts` instantiates the client once and exports the same instance everywhere; This will only need to be addressed when we support multi-chain
- Token client - opened #43, I know Will is refactoring how we get different types of client and I don't want to make his life harder by introducing more changes
- Extensions client - opened #44, this is "blocked" by Will's motion work (nothing else uses extension clients)